lib/platform_dsl.rb

```diff
@@ -95,14 +95,25 @@ def favor_integer_sorting(a, b)
 
     # platform versions may match when the strings do not match, sort order may not be what
     # you expect if you're trying to compare minor versions of EL/SuSE/etc...
+    #
+    # if a platform changes versioning scheme (e.g. macOS 10.X to macOS 11), ignore the missing
+    # version when comparing. this handles the situation where omnibus builds may be tagged with
+    # 11.0 or 11 depending on when they were built (before we fully understood what Apple was
+    # doing with regards to their versioning change).
     def <=>(otherVer)
       raise "comparison between incompatible platform versions:\n#{self}#{otherVer}" if self.mapped_name != otherVer.mapped_name
       if (major_only)
         favor_integer_sorting(self.mapped_major, otherVer.mapped_major)
       else
         ret = favor_integer_sorting(self.mapped_major, otherVer.mapped_major)
-        ret = favor_integer_sorting(self.mapped_minor, otherVer.mapped_minor) if ret == 0 && self.mapped_minor
-        ret = favor_integer_sorting(self.mapped_patch, otherVer.mapped_patch) if ret == 0 && self.mapped_patch
+        if ret == 0 && self.mapped_minor
+          return ret if otherVer.mapped_minor.nil?
```

Comment:
only question i've got here is if its deliberate that "11" also matches "11.2"?  that does seem to work in my head due to yolo mode and partial version mappings, but its not quite called out in the tests that nil now means "match any minor number" rather than just matching zero (at least if I'm also reading the code correctly i suppose).  probably worth a comment and/or a test at any rate.

Reply:
Sure. The intention is that it would support 11.2. I'll add some comments/tests to make that more explicit.